src/main/java/org/gradle/profiler/mutations/CopyFileMutator.java

```diff
@@ -0,0 +1,56 @@
+package org.gradle.profiler.mutations;
+
+import com.google.common.base.Strings;
+import com.google.common.io.Files;
+import com.typesafe.config.Config;
+import org.gradle.profiler.ConfigUtil;
+import org.gradle.profiler.ScenarioContext;
+
+import java.io.File;
+import java.io.IOException;
+
+public class CopyFileMutator extends AbstractFileSystemMutator {
+
+    private final File source;
+    private final File target;
+
+    public CopyFileMutator(File source, File target) {
+        this.source = source;
+        this.target = target;
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    @Override
+    public void beforeScenario(ScenarioContext context) {
+        try {
+            System.out.println("Copying '" + source.getAbsolutePath() + "' to '" + target.getAbsolutePath() + "'");
+            if (!target.exists()) {
+                Files.createParentDirs(target);
+            }
+            Files.copy(source, target);
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to copy '" + source.getAbsolutePath() + "' to '" + target.getAbsolutePath() + "'", e);
```

Comment:
❔ Should we throw UncheckedIOException here, instead of IllegalStateException?

Reply:
You're right.